docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+
+```jldoctest dataframe
+julia> german[:, 3]
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+We can get a single cell of a data frame with the same syntax to get a cell of a matrix:
+
+```jldoctest dataframe
+julia> german[4, 4]
+2
+```
+
+To get a new `DataFrame` that is subset of rows and columns do e.g.:
+
+```jldoctest dataframe
+julia> german[4:5, 4:5]
+2×2 DataFrame
+ Row │ Job    Housing
+     │ Int64  String
+─────┼────────────────
+   1 │     2  free
+   2 │     2  free
+```
+
+## Views
+
+We can create a `view` of a data frame (it is more memory efficient than creating 
+a materialized selection). Here are the possible return value options.
+
+```jldoctest dataframe
+julia> german = copy(german_ref);
+
+julia> s_german1 = view(german, :, 2:5) # column subsetting
+1000×4 SubDataFrame
+  Row │ Age    Sex     Job    Housing
+      │ Int64  String  Int64  String
+──────┼───────────────────────────────
+    1 │    67  male        2  own
+    2 │    22  female      2  own
+    3 │    49  male        1  own
+    4 │    45  male        2  free
+    5 │    53  male        2  free
+    6 │    35  male        1  free
+    7 │    53  male        2  own
+    8 │    35  male        3  rent
+  ⋮   │   ⋮      ⋮       ⋮       ⋮
+  994 │    30  male        3  own
+  995 │    50  male        2  own
+  996 │    31  female      1  own
+  997 │    40  male        3  own
+  998 │    38  male        2  own
+  999 │    23  male        2  free
+ 1000 │    27  male        2  own
+                      985 rows omitted
+```
+
+In this example we used both a function call and a macro:
+
+```jldoctest dataframe
+julia> s_german2 = @view german[end:-1:1, [1, 4]] # row and column subsetting
+1000×2 SubDataFrame
+  Row │ id     Job
+      │ Int64  Int64
+──────┼──────────────
+    1 │   999      2
+    2 │   998      2
+    3 │   997      2
+    4 │   996      3
+    5 │   995      1
+    6 │   994      2
+    7 │   993      3
+    8 │   992      1
+  ⋮   │   ⋮      ⋮
+  994 │     6      2
+  995 │     5      1
+  996 │     4      2
+  997 │     3      2
+  998 │     2      1
+  999 │     1      2
+ 1000 │     0      2
+     985 rows omitted
+```
+
+We will view the element type in vector form:
+
+```jldoctest dataframe
+julia> @view german[1:5, 1]
+5-element view(::Vector{Int64}, 1:5) with eltype Int64:
+ 0
+ 1
+ 2
+ 3
+ 4
+
+julia> @view german[2, 2] # view of the number at 2nd row and 2nd column
+0-dimensional view(::Vector{Int64}, 2) with eltype Int64:
+22
+
+julia> @view german[3, 2:5] # a DataFrameRow, the same as for german[3, 2:5] without a view
+DataFrameRow
+ Row │ Age    Sex     Job    Housing
+     │ Int64  String  Int64  String
+─────┼───────────────────────────────
+   3 │    49  male        1  own
+```
+
+```julia
+julia> @time german[2:5, 2:5]
```

Comment:
what is the purpose of this code block?

Reply:
To show how much memory and time does it take. Previously you had suggested me this.

Reply:
But there is no comment for this code block explaining this.

Reply:
I did not suggest doing this. I suggested:

> comparing timing of `getindex` vs `view` on a large table

Now you are showing `@time` only for `getindex`, and not for `view` and on a small table, where timings are super small and not meaningfull.

Also there is no comment about what the reader can learn from this comparison.